Enable `marketplace-jetpack-plugin-search` feature flag outside production

gcsecsey opened the pull request:
#### Proposed Changes

* Enables the `marketplace-jetpack-plugin-search` feature flag on all environments except production

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

#### On the production environment
* Navigate to `/plugins`
* Open the dev tools
* Perform a search (eg. woocommerce)
* On the dev tools Network tab, validate that a search request was sent out to `https://public-api.wordpress.com/wpcom/v2/marketplace/products` and that the results from the catalog are at the top of the results:
  <img width="1059" alt="CleanShot 2023-01-10 at 13 36 49@2x" src="https://user-images.githubusercontent.com/11555574/211565768-95de2550-e024-41be-873c-055ad7e87cd0.png">

#### On other environments
* Navigate to `/plugins`
* Open the dev tools
* Perform a search (eg. woocommerce)
* On the dev tools Network tab, validate that a search request was sent out to `https://public-api.wordpress.com/rest/v1.3/marketplace/search` and that the results contain free and paid products in a mixed order:
  <img width="1057" alt="CleanShot 2023-01-10 at 13 39 43@2x" src="https://user-images.githubusercontent.com/11555574/211566426-bd3dbbf8-592c-4b01-b10f-117d8f875563.png">

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Closes #63905

gcsecsey committed:
ese onto trunk
enable `marketplace-jetpack-plugin-search` feature flag outside production

epeicher reviewed (approved):
I have tested it, and it works as advertised. Results are mixed in development and in production, all paid plugins are displayed first.

LGTM! 🚢

epeicher commented:
The E2E test failing is trying to locate `Royal Mail` after searching for `shipping` so maybe the test needs to be updated for a more deterministic result for the `shipping` term

gcsecsey commented:
Thanks @epeicher for the review! 🎉 

Yes, I was thinking about updating the tests to a more deterministic result as well, but then realised that Royal Mail should also be a good choice in that regard, and also that the test have been working so far with it. (It has the tag to be relevant for the search and also a fairly high amount of installs) I've tried searching for it directly and found that it only has 1 install currently shown, so the new search ranks it down. 

I've added [this comment](https://github.com/Automattic/wp-calypso/issues/63905#issuecomment-1377470353) about this and raised a diff to resolve the issue. D97495-code Could you please have a quick look at that too? 😄

gcsecsey committed:
update e2e tests

Updates search tests to use Jetpack Protect as Royal Mail is being deranked on results pages

gcsecsey commented:
I've tested the Royal Mail search after reindexing it on my local environment, and the tests now pass.

Nonetheless, I've also updated the E2E tests to use a more relevant search term and result check.

worldomonation commented:
> I've tested the Royal Mail search after reindexing it on my local environment, and the tests now pass.
> 
> Nonetheless, I've also updated the E2E tests to use a more relevant search term and result check.

Thanks for tagging me in the review - I migrated this spec in question from the original Selenium-based iteration.

Royal Mail plugin was used dating back to the old Selenium-based spec and I'm not 100% sure of why, but I am on boarding wit this change as long as the new Jetpack plugin is consistently reliable, which sounds like the case from your comment.

gcsecsey commented:
Thanks for reviewing this @worldomonation!

> Royal Mail plugin was used dating back to the old Selenium-based spec and I'm not 100% sure of why, but I am on board with this change as long as the new Jetpack plugin is consistently reliable, which sounds like the case from your comment.

Yes exactly. We still do expect Royal Mail to show up when the user searches for shipping as it has the tag and is relatively well installed and reviewed. However, we can't guarantee its ranking, especially if the ranking criteria changes in the future.

Searching for the term `jetpack` and looking for any of the Jetpack products should be a more consistent test. Here the name and description also match the search term as well as having many installs and ratings. So Jetpack products should always be featured on the top of the results page.
